### ISSUE TITLE
FAWE is fantastic, but cant cover all

### ISSUE THREAD

**Issue by DR4RCON** [boy0001/FastAsyncWorldedit#656]
Hello!
Dont know if that is correct to report as issue, might be a missing feature or just not possible.
Ive researched for a method to copy a huge amount of blocks from a Forever Stranded World, with lots of mods, about 130. 
Within a lot of popular mods like forestry, harvestcraft, immersive engineering and applied energistics to call some of them (list all down under).

First of all, your mod is awesome fast, compared to everything other have used before. Congratz. And what I wanted to point you to here, cant be done by any other editor mod. With worldedit only and also with mcedit2, only the try to copy it crashes 9 from 10 times and not half of your result is reached.
I also coding and I know its not only a tricky job, to cover all different technics developer uses over all that mods from that huge community, but an impossible.

But if my report helps only to cover some more of that famous mods, I will provide you any information you need to make your stuff better.

Forge Mod Loader version 12.18.3.2316 for Minecraft 1.10.2 loading
Java is Java HotSpot(TM) 64-Bit Server VM, version 1.8.0_131, running on Linux:amd64:3.16.0-4-amd64, installed at /usr/lib/jvm/java-8-oracle/jre

Mods:

appliedenergistics2{rv4-alpha-11}
actuallyadditions{1.10.2-r105}
advancedRocketry{1.2.2}
architecturecraft{1.7.1}
AutoRegLib{1.0-2}
base{1.10.2-1.4.0}
battletowers{1.5.7}
Baubles{1.3.11}
bdlib{1.12.4.24}
betterquesting{2.3.231}
betterwithmods{0.14.24 Beta}
bigreactors{1.10.2-0.4.5.29}
bloodmoon{1.5.1}
bq_standard{2.3.131}
bqt{0.9}
ccl-entityhook{1.0}
Chameleon{1.10.2-2.3.0}
ChickenChunks{2.2.0.52}
chisel{MC1.10.2-0.0.10.22}
chiselsandbits{12.12}
clumps{1.0.5}
CodeChickenCore{2.4.1.102}
CodeChickenLib{2.5.8.255}
cofhcore{4.1.1}
configmod{v1.0}
cookingforblockheads{4.2.43}
crafttweakerjei{1.0.1}
elevatorid{1.3.0}
endercore{1.10.2-0.4.1.66-beta}
EnderIO{1.10.2-3.1.192}
EnderStorage{2.2.1.106}
environmentaltech{1.10.2-0.10.6b}
excompressum{2.0.87}
exnihiloomnia{1.2.6}
extendedrenderer{v1.0}
extrautils2{1.0}
flatcoloredblocks{mc1.10-v4.4}
forestry{5.2.17.376}
forgemultipartcbe{2.0.0.31}
gravestone{1.5.12}
harvestcraft{1.10.2j}
hopperducts{1.4.8}
IC2{2.6.210-ex110}
immersiveengineering{0.10-58}
industrialwires{1.4-10}
infinitefluids{1.1.0}
ironbackpacks{1.10.2-2.2.27}
ironchest{1.10.2-7.0.15.804}
libVulpes{0.2.5.}
mantle{1.10.2-1.1.4.203}
microblockcbe{2.0.0.31}
mikedongles{0.4.4-[1.9.41.101.10.2]}
minecraftmultipartcbe{2.0.0.31}
MineTweaker3{3.0.25}
modtweaker{2.0.12}
mpbasic{1.1.3}
mputils{1.2.9}
mtlib{@VERSION@}
omlib{3.0.0-101}
openglider{@VERSION@}
openmodularturrets{3.0.0-158}
pressure{1.3.1.25}
progressiveautomation{1.7.4}
props{2.4.2}
PTRModelLib{1.0.0}]
quantumstorage{3.3.4}
Quark{r1.1-70}
railcraft{10.1.2}
randomthings{3.7.7.1}
reborncore-mcmultipart{2.13.6.142}
reborncore{2.13.6.142}
redstonearsenal{2.0.3}
simplegenerators{1.10.2-0.10.6a}
simplyjetpacks{2.1.1.47}
slug{1.0A}
snad{1.10.2-1.6.09.20a}
soulshardstow{1.10.2-2.6.7-48}
SpiceOfLife{1.3.8}
stats_keeper{1.4.5}
StorageDrawers{1.10.2-3.7.8}
supercraftingframe{1.2.1.88}
tanaddons{1.0.4}
tconstruct{1.10.2-2.6.2.487}
thermalexpansion{5.0.4}
thermalfoundation{2.0.5}
thirstybottles{1.0.1.0}
TinkersAddons{1.0.5}
torchmaster{1.0}
ToughAsNails{@MOD_VERSION@}
tp{1.2.0-Hotfix}
valkyrielib{1.10.2-0.10.6}
wct{0.1.31}
weather2{1.10.2-2.4.10}
Woot{1.10.2-1.4.0}
zerocore{1.10.2-0.1.0.6}
zombieawareness{1.10.2-1.11.7}

Finally:
worldedit-forge-mc1.10.2-6.1.4
FastAsyncWorldEdit-forge110-17.07.24-9649482-760-13.7.18

Method to reproduce:
1) Used wand to set first and second position. From right front to left behind.
2) Expanded vertical ( //expand vertical )
3) Standing in front of the main entrance, Copied ( //copy )
4) 300 blocks left of it, pasted. ( //paste )

Copied about 20.000.000 Blocks 

What Ive observed so far, is:
1) EnderIO Blocks copied without texture. Some models are there, but transparent.
2) Applied Energistics 2 cable are not copied, some are, but without textures. ME Drives are not copied. 
3) Immersive Engineering Multiblocks are with wrong orientation, all blocks seems to copied, but often with broken textures.
4) Advanced Rocketry blocks are all copied, but the multiblock structure is broken. Trying to reinitilize the multiblock structure, crashes the instance.
5) Ex Compressum loses Metadata for example Auto Sieves and Auto Hammer
6) Some positive, BetterWithMods worked directly after copying without any intervention.
7) Chisel and Factory seems to be copied as well.
8) Woot mob factory and blocks copied also
9) Harvestcraft multiblocks works too

Hope that helps to improve your fantastic editor mod. Our world is still in progress, but thinking to the future we would like to continue with that schematic in another world. Hope could copy more then.

When I can provide more information let me know.
Thanks,
Regards Drarcon

![2017-07-24_14 46 22](https://user-images.githubusercontent.com/30311920/28525123-782bd860-7083-11e7-9739-b1593fbf8b32.png)
![2017-07-24_14 47 09](https://user-images.githubusercontent.com/30311920/28525121-7828a528-7083-11e7-98bf-28b1dcdbba13.png)
![2017-07-24_14 48 09](https://user-images.githubusercontent.com/30311920/28525124-78306bbe-7083-11e7-83e2-3e3d3905db7e.png)
![2017-07-24_14 48 48](https://user-images.githubusercontent.com/30311920/28525122-78292f16-7083-11e7-99bd-cb5bfe98e927.png)
![2017-07-24_14 49 56](https://user-images.githubusercontent.com/30311920/28525126-7848ecd4-7083-11e7-86d8-8e9aeae3c5c2.png)
![2017-07-24_14 50 47](https://user-images.githubusercontent.com/30311920/28525127-78494f8a-7083-11e7-9eb4-e15d04d1c535.png)
![2017-07-24_14 52 18](https://user-images.githubusercontent.com/30311920/28525125-7841c134-7083-11e7-9de0-9b1123abb099.png)
![2017-07-24_14 53 07](https://user-images.githubusercontent.com/30311920/28525128-784e193e-7083-11e7-933a-9201cec768ab.png)
![2017-07-24_14 54 10](https://user-images.githubusercontent.com/30311920/28525131-7867550c-7083-11e7-9c13-1f0c931e4b0e.png)
![2017-07-24_14 55 13](https://user-images.githubusercontent.com/30311920/28525132-7868d12a-7083-11e7-80f4-894fe0af3c2c.png)
![2017-07-24_14 57 57](https://user-images.githubusercontent.com/30311920/28525129-785f15a4-7083-11e7-94ed-5eda668b34d4.png)
![2017-07-24_14 58 02](https://user-images.githubusercontent.com/30311920/28525133-786ba6de-7083-11e7-928b-ec3f4ac7e42f.png)
![2017-07-24_14 58 47](https://user-images.githubusercontent.com/30311920/28525135-787e93c0-7083-11e7-8925-b9981fe2114b.png)

**Comment by FrankHeijden** [comment]
offtopic question; what is your average TPS with so many mods o.o

**Comment by DR4RCON** [comment]
As you can see about 30-40 fps at really complex views. So its playable!

**Comment by dooferorg** [comment]
Apologies for the necropost but has there been a solution to this? Using Forge  on MC1.12

I'm having that same issue with the latest release of FAWE and some ProjectRed mod blocks, namely lights. It seems the variant IDs are not copied. In my instance the white lamp is ID #6942:0 .. I have inverted white lamps that I copied which are #6942:16 .. when pasted the blocks are #6942:0 

So it seems the variant part is what's being missed off the paste operation. That would also explain things like stairs etc. as well
